### ISSUE TITLE
Python 3.6 wheel

### ISSUE THREAD

**Issue by robambalu** [finos/perspective#1067]
We're interested in using python-perspective but we're currently bound to Python 3.6 ( we're on 3.6.4 to be precise )
Would it be possible to get a 3.6 wheel out?

**Comment by timkpaine** [comment]
@robambalu can probably do a one-off for experimentation, can you provide platform specs?

**Comment by texodus** [comment]
We are making good progress on wheels, it is under active development right now e.g. #1057.  

I'll be sure to update this Issue with details as we release these.  It's probably also a good idea for us to update the Python docs to reflect the current wheel support per OS/Python Version, so I'll get that added to.

**Comment by timkpaine** [comment]
the wheels in #1057 won't help for python3.6 though. we have no plans to support any versions < 3.7 beyond end of 2020

**Comment by robambalu** [comment]
Hey guys! been a while!
Our current build is on Linux RHEL7, 64-bit Python 3.6.4
Any other information you need?

**Comment by BarCatFigaro** [comment]
Hi guys, we'd also be interested in a 3.6 wheel, if possible!

We're currently on CentOS 7 (64-bit) and running Python 3.6.3.

Let me know if there is any other information required!

**Comment by sneakychuhcu** [comment]
Hello, I was also interested in this feature; when will it be available? Thank you!

**Comment by robambalu** [comment]
@timkpaine any update on that build? Should we expecting it or is it not happening any time soon

**Comment by BarCatFigaro** [comment]
@timkpaine it seems like the PR is just about to be wrapped up; would I be able to build a 3.6 wheel myself if I cloned the repo and branch, or would it require the changes to be merged in? I'm working on spinning up a prototype and would love to get started with the python runtime! Thanks :)

**Comment by BarCatFigaro** [comment]
As a follow up, I managed to build the 3.6 wheel and will work with it for the time being.

**Comment by timkpaine** [comment]
this is done, we are still working out some details about publishing to pypi but the source dist should build across platforms py2.7/py3.6/py3.7/py3.8 and their are some wheels available https://dev.azure.com/finosfoundation/perspective/_build/results?buildId=912&view=artifacts&type=publishedArtifacts (though ymmv due to ongoing manylinux issues)

**Comment by sc1f** [comment]
Currently working on fixing the manylinux wheels for Python 3 but will require a docker image re-dist. To make the manylinux wheels from Azure artifacts work:

1. after installing perspective and attempting to import, you'll get an error that says "unable to import C++ bindings"
2. go to your site-packages folder/wherever you installed, and there should be a folder next to perspective called `perspective_python.libs`
3. go to `perspective/table`, and copy `libpsp.so` and `libtbb.so` into the `perspective_python.libs` folder. This should fix the import error.

Feel free to open an issue if that process doesn't work/y'all hit any other issues - the more visibility the better

**Comment by robambalu** [comment]
awesome, thanks fellas!

**Comment by adpoint72** [comment]
where can i get python 3.6 wheel?

**Comment by robambalu** [comment]
@sc1f did you get anywhere with fixing the manylinux wheels?  Its still unclear where we can get the 3.6 version
I also dont see a 3.6 wheel in that azure link @timkpaine

**Comment by sc1f** [comment]
The PR for fixing manylinux wheels to just "work" is still open: #1057 

I plan on taking another look at it very soon - verifying all the wheels are clean takes a little time.

You can use our CI-generated wheels from the 0.5.2 release, although you will need to follow the instructions in [my comment](https://github.com/finos/perspective/issues/1067#issuecomment-649762785) above. Afterwards it should just work:

https://dev.azure.com/finosfoundation/perspective/_build/results?buildId=1035&view=artifacts&type=publishedArtifacts

We will not be publishing these wheels to PyPi until the Manylinux build is fixed - hoping to get that done in the next few days.

**Comment by adpoint72** [comment]
thanks for quick reply.
i see 2 3.6 wheels, what's the difference?
cp36-cp36m-manylinux2010_x86_64
cp36-cp36m-manylinux2014_x86_64

**Comment by sc1f** [comment]
@adpoint72 https://uwekorn.com/2019/09/15/how-we-build-apache-arrows-manylinux-wheels.html

TLDR: depends on your system. Manylinux2014 will work on CentOS7 and above, Manylinux2010 will work on CentOS6 and above. Or download both and try.

**Comment by adpoint72** [comment]
we are on 
CentOS Linux release 7.7.1908 (Core)
Python 3.6.4
I tried both 3.6 wheels and get the following when trying to install
perspective_python-0.5.2-cp36-cp36m-manylinux2010_x86_64.whl is not a supported wheel on this platform.
perspective_python-0.5.2-cp36-cp36m-manylinux2014_x86_64.whl is not a supported wheel on this platform.

**Comment by sc1f** [comment]
@adpoint72 what is your output from `python -c "from pip import pep425tags;print(pep425tags.supported_tags)"`? You may also have to update pip so it recognizes the manylinux2010/2014 tags.

**Comment by adpoint72** [comment]
thanks @sc1f, upgrading pip helped.
i see that installing that wheel downgraded our arrow from 0.17.1 to 0.16.0.  
anyway to avoid that?

**Comment by timkpaine** [comment]
No we have a arrow C++ dependencies so the requirements are 0.15.1 or 0.16.0 for now
